.github/workflows/ci.yaml

```diff
@@ -0,0 +1,354 @@
+name: CI
+
+on:
+  push:
+    branches: [main]
+  merge_group:
+  pull_request:
+    branches:
+      - "main"
+      - "v[0-9]+.[0-9]+.[0-9]+"
+      - "v[0-9]+.[0-9]+"
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-20.04 
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v4
+
+    - name: Verify
+      uses: golangci/golangci-lint-action@v3
+      with:
+        version: v1.51.2
+        args: --timeout=15m0s --verbose
+  
+  unit:
+    runs-on: ubuntu-20.04
+    name: Unit and Integration tests
+    steps:
+    - name: Checkout
```

Comment:
this is already part of the setup action

Reply:
I removed "Checkout" from the Setup

Reply:
We talked and decided not to change.